cmd/agent/app/reporter/tchannel/collector_proxy.go

```diff
@@ -25,22 +25,28 @@ import (
 
 // ProxyBuilder holds objects communicating with collector
 type ProxyBuilder struct {
-	reporter reporter.Reporter
+	reporter *reporter.ClientMetricsReporter
 	manager  configmanager.ClientConfigManager
 	tchanRep *Reporter
 }
 
 // NewCollectorProxy creates ProxyBuilder
 func NewCollectorProxy(builder *Builder, mFactory metrics.Factory, logger *zap.Logger) (*ProxyBuilder, error) {
```

Comment:
a little bit irrelevant to data loss, we don't need to correct it in this PR. But usually Builder.NewXXX means build a new XXX,  seems more reasonable for NewCollectorProxy to create a CollectorProxy instead of ProxyBuilder.

Reply:
I agree, the naming here has become weird overtime.